Client/Frontend/Browser/BrowserViewController.swift

```diff
@@ -2483,11 +2494,26 @@ extension BrowserViewController: ToolbarDelegate {
     }
     
     func tabToolbarDidPressMenu(_ tabToolbar: ToolbarProtocol) {
-        let homePanel = MenuViewController(bvc: self, tab: tabManager.selectedTab)
-        let popover = PopoverController(contentController: homePanel, contentSizeBehavior: .preferredContentSize)
-        // Not dynamic, but trivial at this point, given how UI is currently setup
-        popover.color = Theme.of(tabManager.selectedTab).colors.home
-        popover.present(from: tabToolbar.menuButton, on: self)
+        let selectedTabURL: URL? = {
+            guard let url = tabManager.selectedTab?.url, !url.isLocal || url.isReaderModeURL else { return nil }
+            return url
+        }()
+        var activities: [UIActivity] = []
+        if let url = selectedTabURL, let tab = tabManager.selectedTab {
+            activities = shareActivities(for: url, tab: tab, sourceView: view, sourceRect: self.view.convert(self.topToolbar.menuButton.frame, from: self.topToolbar.menuButton.superview), arrowDirection: .up)
+        }
+        let menuController = MenuViewController(content: { menuController in
+            VStack(spacing: 6) {
```

Comment:
do you think we should start adding these values in struct or unnecessary for single instance ?

Reply:
I think it is unnecessary at the moment, as it is a small computed View. If the menu grows to be larger though we can always reconsider